src/meta/src/hummock/manager/mod.rs

```diff
@@ -995,36 +969,32 @@ where
                     start_time.elapsed(),
                     compact_task_statistics
                 );
-        }
-
-        #[cfg(test)]
-        {
-            drop(compaction_guard);
-            self.check_state_consistency().await;
+        } else {
+            // For trivial tasks, commit the memory state for subsequent reports.
+            compact_task_assignment.commit();
```

Comment:
Because trivial task shall not change `compact_status`.

Reply:
@hzxa21  This point also confuse me. 

The trivial task is added to the task assignment in order to take it out when reporting the task. Since we are no longer passing the entire compact task structure